src/Microsoft.TemplateEngine.Cli/HelpAndUsage/HelpForTemplateResolution.cs

```diff
@@ -423,51 +399,84 @@ public static CreationResultStatus HandleParseError(INewCommandInput commandInpu
+            {
+                if (inputParametersList.Length != 0)
+                {
+                    inputParametersList.Append(separator);
+                }
+                inputParametersList.AppendFormat("type='{0}'", context);
             }
-            if (!string.IsNullOrEmpty(context)){
-                inputParametersList.Add("type=" + context);
+            if (!string.IsNullOrEmpty(baselineName))
+            {
+                if (inputParametersList.Length != 0)
+                {
+                    inputParametersList.Append(separator);
+                }
+                inputParametersList.AppendFormat("baseline='{0}'", baselineName);
             }
-            return String.Join(", ", inputParametersList);
+            return inputParametersList.ToString();
         }
 
-        private static void ShowNoTemplatesFoundMessage(string templateName, string templateLanguage, string context)
+        private static void ShowNoTemplatesFoundMessage(string templateName, string templateLanguage, string context, string baselineName)
         {
             // No templates found matching the following input parameter(s): {0}.
-            Reporter.Error.WriteLine(string.Format(LocalizableStrings.NoTemplatesMatchingInputParameters, GetInputParametersString(templateName, templateLanguage, context)).Bold().Red());
+            Reporter.Error.WriteLine(string.Format(LocalizableStrings.NoTemplatesMatchingInputParameters, GetInputParametersString(templateName, templateLanguage, context, baselineName)).Bold().Red());
         }
 
-        private static void ShowTemplatesFoundMessage(string templateName, string templateLanguage, string context)
+        private static void ShowTemplatesFoundMessage(string templateName, string templateLanguage, string context, string baselineName)
         {
-            if (!string.IsNullOrEmpty(templateName) || !string.IsNullOrEmpty(templateLanguage) || !string.IsNullOrEmpty(context)){
+            if (!string.IsNullOrEmpty(templateName) || !string.IsNullOrEmpty(templateLanguage) || !string.IsNullOrEmpty(context) || !string.IsNullOrEmpty(baselineName))
+            {
                 // Templates found matching the following input parameter(s): {0}
-                Reporter.Output.WriteLine(string.Format(LocalizableStrings.TemplatesFoundMatchingInputParameters, GetInputParametersString(templateName, templateLanguage, context)));
+                Reporter.Output.WriteLine(string.Format(LocalizableStrings.TemplatesFoundMatchingInputParameters, GetInputParametersString(templateName, templateLanguage, context, baselineName)));
                 Reporter.Output.WriteLine();
             }
         }
+
+        private static string GetPartialMatchReason(ListOrHelpTemplateListResolutionResult templateResolutionResult, string templateLanguage, string context, string baselineName)
+        {
+            StringBuilder reason = new StringBuilder();
+            string separator = ", ";
```

Comment:
should this be a global static value?

Reply:
I don't think this separator is used globally, however it is the same at the moment for GetPartialMatchReason and GetInputParametersString.  
I would prefer to keep it separate though, if pattern change for one of the messages; as alternative we can create the global collection of static separators and use them when needed. 
@AR-May are you aware if we already have a class containing global constant strings we are using in the CLI, I don't think I saw one

Reply:
I have not seen such a class also.